[composite compliance] Fix a bunch of fft backwards

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #76574
* __->__ #76573

Replaced `at::zeros(..., grad.options()).slice().copy_(grad))`
with `grad.new_zeros(..., grad.options()).slice().copy_(grad))`

Test Plan:
- run tests

zou3519 committed:
[composite compliance] Fix a bunch of fft backwards

Replaced `at::zeros(..., grad.options()).slice().copy_(grad))`
with `grad.new_zeros(..., grad.options()).slice().copy_(grad))`

Test Plan:
- run tests

[ghstack-poisoned]

albanD reviewed (approved):
SGTM
There is a missing xfail that needs to be removed to appease CI though.

zou3519 committed:
Update on "[composite compliance] Fix a bunch of fft backwards"


Replaced `at::zeros(..., grad.options()).slice().copy_(grad))`
with `grad.new_zeros(..., grad.options()).slice().copy_(grad))`

Test Plan:
- run tests

[ghstack-poisoned]
